gpu/kinfu_large_scale/tools/kinfuLS_app.cpp

```diff
@@ -977,29 +981,59 @@ struct KinFuLSApp
   }
 
   void
-  startMainLoop (bool triggered_capture)
+  startMainLoop (bool triggered_capture, bool oni2_interface = false)
   {   
-    using namespace openni_wrapper;
-    using DepthImagePtr = boost::shared_ptr<DepthImage>;
-    using ImagePtr = boost::shared_ptr<Image>;
+    boost::signals2::connection c;
 
-    std::function<void (const ImagePtr&, const DepthImagePtr&, float)> func1 = [this] (const ImagePtr& img, const DepthImagePtr& depth, float constant)
-    {
-      source_cb2 (img, depth, constant);
-    };
-    std::function<void (const DepthImagePtr&)> func2 = [this] (const DepthImagePtr& depth) { source_cb1 (depth); };
-    std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&) > func3 = [this] (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& cloud)
+    if(oni2_interface)
     {
-      source_cb3 (cloud);
-    };
+      using namespace pcl::io;
+      using DepthImagePtr = boost::shared_ptr<DepthImage>;
+      using ImagePtr = boost::shared_ptr<Image>;
+
+      std::function<void (const ImagePtr&, const DepthImagePtr&, float)> func1 = [this] (const ImagePtr& img, const DepthImagePtr& depth, float constant)
```

Comment:
`const auto name = [](){}` is a more terse syntax for lambda, and it'll still cast itself automatically to std::function

Reply:
I agree the syntax is way too much verbose but casting a lambda to an `std::function<>` is not straightforward for me.

for example, replacing `func2` declaration with 

```
      const auto func2 = [this] (const DepthImagePtr& depth) { source_cb1 (depth); };
```

raise the error : 

```
/home/antoine/dev/pcl/io/include/pcl/io/grabber.h:246:12: note: candidate template ignored: could not match 'function<type-parameter-0-0>' against 'const (lambda at /home/antoine/dev/pcl/gpu/kinfu_large_scale/tools/kinfuLS_app.cpp:1010:26)'
```

and as far as I understand the `std::function` passed to `Grabber::registerCallBack` should be fully typed. It can't deduce the type from the lambda.
See https://stackoverflow.com/questions/13358672/how-to-convert-a-lambda-to-an-stdfunction-using-templates

Reply:
> It can't deduce the type from the lambda

Sort of. `Grabber::registerCallback` being templated is the issue since now the function wants the lambda to fit the type and the lambda wants a type before it can be implicitly converted. If the `registerCallback` is [instantiated](https://godbolt.org/z/LAaOc7), it's not an issue (but doing that negates using lambda). Leave it be :(